src/libfuncs/uint64.rs

```diff
@@ -330,246 +307,175 @@ pub fn build_square_root<'ctx, 'this>(
     let i32_ty = IntegerType::new(context, 32).into();
     let i64_ty = IntegerType::new(context, 64).into();
 
-    let k1 = entry
-        .append_operation(arith::constant(
-            context,
-            IntegerAttribute::new(i64_ty, 1).into(),
-            location,
-        ))
-        .result(0)?
-        .into();
-
-    let is_small = entry
-        .append_operation(arith::cmpi(
-            context,
-            CmpiPredicate::Ule,
-            entry.argument(1)?.into(),
-            k1,
-            location,
-        ))
-        .result(0)?
-        .into();
-
-    let result = entry
-        .append_operation(scf::r#if(
-            is_small,
-            &[i64_ty],
-            {
-                let region = Region::new();
-                let block = region.append_block(Block::new(&[]));
-
-                block.append_operation(scf::r#yield(&[entry.argument(1)?.into()], location));
-
-                region
-            },
-            {
-                let region = Region::new();
-                let block = region.append_block(Block::new(&[]));
-
-                let k64 = entry
-                    .append_operation(arith::constant(
+    let k1 = entry.append_op_result(arith::constant(
```

Comment:
you can create a constant in an easier way using the new API (`const_int_from_type` or `const_int`)

Reply:
yes just saw that, fixed